var/spack/repos/builtin/packages/argobots/package.py

```diff
@@ -20,6 +20,7 @@ class Argobots(AutotoolsPackage):
     git      = "https://github.com/pmodels/argobots.git"
 
     version("develop", branch="master")
+    version("1.0rc1", "729c018f3353976cb79c07ecaa13319d")
```

Comment:
Because of the unusual (non-numeric) version numbers here... I would make sure that, left to its own devices, Spack picks the version you are intending (i.e. the latest one).  Try `spack spec argobots` on a "clean" installation (i.e. no YAML files to direct it), and see what happens.

Reply:
I tested this using the `spack python` console and the commands listed [here](https://spack.readthedocs.io/en/latest/developer_guide.html#id3) (adjusted to this problem). It does work: 1.0rc1>1.0b1 > 1.0a1. It makes sense based on the code that I found [here](https://github.com/spack/spack/blob/develop/lib/spack/spack/version.py#L275-L294). It works since the numeric part of the versions are the same and the string version is also ordered: 'a' < 'b' < 'rc' .

Reply:
```
> spack spec margo
Input spec
--------------------------------
margo

Concretized
--------------------------------
margo@0.4.3%gcc@7.3.0+shared arch=linux-ubuntu18.04-x86_64
    ^argobots@1.0rc1%gcc@7.3.0~valgrind arch=linux-ubuntu18.04-x86_64
...
```

```
> spack spec argobots
Input spec
--------------------------------
argobots

Concretized
--------------------------------
argobots@1.0rc1%gcc@7.3.0~valgrind arch=linux-ubuntu18.04-x86_64
```

Reply:
> I tested this using..

Thanks for checking, I think we got lucky this time ;-)